test/testcondition.cpp

```diff
@@ -3631,6 +3633,84 @@ class TestCondition : public TestFixture {
               "}\n");
         ASSERT_EQUALS("", errout.str());
     }
+
+    void smartpointer() {
+        check("#include <memory>\n"
+              "\n"
+              "bool h();\n"
+              "\n"
+              "class C\n"
+              "{\n"
+              "public:\n"
+              "  void f();\n"
+              "  void g();\n"
+              "\n"
+              "  std::shared_ptr<int> x;\n"
+              "};\n"
+              "\n"
+              "void C::g()\n"
+              "{\n"
+              "  if (h()) {\n"
+              "    x.reset(new int);\n"
+              "  }\n"
+              "}\n"
+              "\n"
+              "void C::f()\n"
+              "{\n"
+              "  x.reset();\n"
+              "  g();\n"
+              "  if (!x) {\n"
+              "  }\n"
+              "}\n");
+        ASSERT_EQUALS("", errout.str());
+
+        check("#include <memory>\n"
+              "\n"
+              "class C\n"
+              "{\n"
+              "public:\n"
+              "  void f();\n"
+              "  void g();\n"
+              "\n"
+              "  std::shared_ptr<int> x;\n"
+              "};\n"
+              "\n"
+              "void C::g()\n"
+              "{\n"
+              "    x.reset(new int);\n"
+              "}\n"
+              "\n"
+              "void C::f()\n"
+              "{\n"
+              "  x.reset();\n"
+              "  g();\n"
+              "  if (!x) {\n"
+              "  }\n"
+              "}\n");
+        TODO_ASSERT_EQUALS("[test.cpp:18]: (style) Condition '!x' is always false\n", "", errout.str());
+
+        check("#include <memory>\n"
+              "\n"
+              "class C\n"
+              "{\n"
+              "public:\n"
+              "  void f();\n"
+              "  void g();\n"
+              "\n"
+              "  std::shared_ptr<int> x;\n"
+              "};\n"
+              "\n"
+              "void C::g(){}\n"
+              "\n"
+              "void C::f()\n"
```

Comment:
Nit; I don't have a strong opinion but putting the `{` on the same line makes the test case a bit more compact.

We should not make test cases extremely compact but one goal is that it will be possible to compare test cases and get a bigger picture about what is tested.